### DISCUSSION TITLE
Brownie does not recognise local ganache automatically.

### DISCUSSION THREAD

**Discussion by akhilparakka** [smartcontractkit/full-blockchain-solidity-course-py#1607] [Q&A] [answered]
![Screenshot from 2022-07-27 10-28-20](https://user-images.githubusercontent.com/77268333/181164903-1f17a1e8-be21-4518-89d5-279535c6b2b3.png)

**Comment by cromewar** [comment]
Hello @Akhil-franko 
Could you explain in more detail the issue?
Please give a check to #980

**Reply by akhilparakka** [reply]
It was said in the tutorial that if the ganache application is running, brownie will automatically detect that application. But when i try to run deploy.py, brownie creates a ganache-cli in the terminal.

**Reply by fantauzzi** [reply]
Ensure the ganache application you are running is listening on the same port that brownie is connecting to. You can check which port brownie is connecting to by running your `deploy.py`; in the output you get something like 

`Launching 'ganache-cli --port 8545 --gasLimit 12000000 --accounts 10 --hardfork istanbul --mnemonic brownie'...
`
In my case you see the port is 8545. If I run ganache listening on that same port, brownie will not start another instance of ganache but connect to the running one instead.

If you are using `ganache-cli` you can set the port with the `--port` parameter, if you are running the Ganache GUI, you can set the port in its settings, under the SERVER tab.

**Reply by akhilparakka** [reply]
The port mentioned under the server tab in Settings was 7545. Changed it to 8545. brownie detected Ganache GUI.
Thank You.

**Comment by akhilparakka** [comment, marked as answer]
The port mentioned under the server tab in Settings was 7545. Changed it to 8545. brownie detected Ganache GUI.
Thank You.
